### PR TITLE
Pre commit compliance

### PR DESCRIPTION
**Pull request by KPostOffice**
Use pre-commit and address mypy and other errors that were pre-existing

### CODE REVIEW

File: thoth/adviser/predictors/latest.py [hunk 1]
```diff
@@ -41,24 +42,20 @@ class ApproximatingLatest(HillClimbing):
...
-    _hop = attr.ib(type=bool, default=False)
-    _hop_logged = attr.ib(type=bool, default=False)
-    _packages_heated_up = attr.ib(type=Set[str], factory=set)
-    _initial_state = attr.ib(type=Optional[State], default=None)
-    _latest_versions_heat_up = attr.ib(type=set, factory=set)
+    _hop = attr.ib(type=bool, default=False, init=False)
+    _hop_logged = attr.ib(type=bool, default=False, init=False)
+    _packages_heated_up = attr.ib(type=Set[str], factory=set, init=False)
+    _initial_state = attr.ib(type=Optional[State], default=None, init=False)
+    _latest_versions_heat_up = attr.ib(type=Set[Any], factory=set, init=False)
```
Comment (fridex) [inline thoth/adviser/predictors/latest.py:49]: Why `init=False` here and other places?

Comment (KPostOffice) [reply thoth/adviser/predictors/latest.py:49]: Because it shouldn't be a keyword or positional argument. They are only set by internal functions

**Comment by KPostOffice** [comment]
> Where there done any changes that would affect computing?

@fridex There were a few `None` checks but that's all

**Pull request merged**